src/api/lib/backend/api/build_results/status.rb

```diff
@@ -47,8 +47,9 @@ def self.result_swiss_knife(project_name, params = {})
 
         # Lists failed package builds in a project
         # @return [String]
-        def self.failed_results(project_name)
-          response = http_get(['/build/:project/_result', project_name], params: { code: %w[failed broken unresolvable] }, expand: [:code])
+        def self.failed_results(project_name, params = {})
```

Comment:
I'm only taking forward other's ideas, but this is not what was intended.

Rather an extra `self.failed_result_summary_for_repo(project_name, repository_name)`

what you're doing here is tempting, but is just swiss armying another backend function.

Reply:
If the plan is to have individual method calls for certain query types, eg. one for '/build/:project/_result' and query parameter `view: :status` and one for '/build/:project/_result' and query parameter `view: :summary`, I can understand that.

But for every combination of API query parameters? That seems to much. For '/build/:project/_result' there would be architecture, repository, package, code, view and maybe more.
In the end we would end up adding a new layer, which wraps these method calls per API endpoints. Because for each API endpoint they would only differ by their params.

Reply:
Well, the idea (and again, it's not mine, I'm argueing here for the sake of consistency) is to abstract how the backend wants it from how the API wants it. If you put `code: 'summary'` in your model, you're putting 'how the backend wants it' logic in API code.

And in the end, the backend does different things for different views, so having 2 functions in lib/backend doesn't sound that wrong.

Reply:
> to abstract how the backend wants it from how the API wants it.

That sounds like a good plan.

> If you put code: 'summary' in your model, you're putting 'how the backend wants it' logic in API code.

No, that's how we want the data for this particular case: The minimal data to know if a package build failed, aka. matches 3 specific codes.

> And in the end, the backend does different things for different views, so having 2 functions in lib/backend doesn't sound that wrong.

The devil is in the detail. If I want an index view, with a query parameter (aka. filter), you don't need two different endpoints for this.
And in this case I want the build results for a specific project filtered by their `code` (being unresolvable, broken or failed). 

If you meant `view: 'summary` and not `code` then I already agreed with that one:
> If the plan is to have individual method calls for certain query types, eg. one for '/build/:project/_result' and query parameter view: :status and one for '/build/:project/_result' and query parameter view: :summary, I can understand that.

Reply:
I changed parameter type from a hash to plain variables. Though I didn't add the `_for_repo` suffix to  the method name as suggested, since none of the others do that.